.github/workflows/cryostat-build.yml

```diff
@@ -0,0 +1,84 @@
+on:
+  workflow_call:
+    inputs:
+      build-arch:
+        required: true
+        type: string
+      skip-itests:
+        required: false
+        type: boolean
+        default: false
+    outputs:
+      image-version:
+        description: the Cryostat application version that will be built
+        value: ${{ jobs.get-pom-properties.outputs.image-version }}
+
+env:
+  CURRENT_VERSION: "2.24.0"
+  TEST_VERSION: 2.25.0-SNAPSHOT
+
+jobs:
+  get-pom-properties:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: ${{ github.repository_owner }}/cryostat
+        ref: main
+    - uses: Mudlet/xmlstarlet-action@master
+      with:
+        args: ed -P -L -u '//_:properties[_:io.cryostat.core.version=${{ env.CURRENT_VERSION }}]/_:io.cryostat.core.version' -v ${{ env.TEST_VERSION }} pom.xml 
```

Comment:
You can actually do this even more simply, without modifying the `pom.xml`:

```bash
$ ./mvnw -Dio.cryostat.core.version=${{ env.TEST_VERSION}} clean package
```

Reply:
This way you also don't need the `CURRENT_VERSION` env var because you're letting Maven directly override the property value instead of doing a text-based substitution.